src/components/dialogs/restore-node-dialog.js

```diff
@@ -0,0 +1,146 @@
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { FormattedMessage, useIntl } from 'react-intl';
+import {
+    Box,
+    DialogContentText,
+    FormControlLabel,
+    Radio,
+    RadioGroup,
+} from '@mui/material';
+import {
+    fetchStashedNodes,
+    restoreStashedNodes,
+} from '../../services/study/tree-subtree';
+import LoaderWithOverlay from '../utils/loader-with-overlay';
+import FormControl from '@mui/material/FormControl';
+
+/**
+ * Dialog to select network modification to create
+ * @param {Boolean} open Is the dialog open ?
+ * @param {EventListener} onClose Event to close the dialog
+ * @param anchorNodeId the anchor node id relative to which the restore will take place
+ * @param studyUuid the study id
+ */
+const RestoreNodesDialog = ({ open, onClose, anchorNodeId, studyUuid }) => {
+    const [nodes, setNodes] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [selectedNode, setSelectedNode] = useState(null);
+    const intl = useIntl();
+
+    useEffect(() => {
+        if (open) {
+            setIsLoading(true);
+            fetchStashedNodes(studyUuid).then((res) => {
+                setNodes(res);
+                setIsLoading(false);
+            });
+        } else {
+            setNodes([]);
+            setIsLoading(false);
+        }
+    }, [studyUuid, open]);
+
+    const handleClose = () => {
+        onClose();
+        setSelectedNode(null);
+        setIsLoading(false);
+        setNodes([]);
+    };
+
+    const handleRestore = () => {
+        restoreStashedNodes(studyUuid, selectedNode.id, anchorNodeId);
+        handleClose();
+    };
+
+    return (
+        <Dialog
+            fullWidth
+            maxWidth="xs"
+            open={open}
+            onClose={handleClose}
+            aria-labelledby="dialog-restore-nodes"
+        >
+            <DialogTitle>
+                {intl.formatMessage({
+                    id: 'restoreNodes',
+                })}
+            </DialogTitle>
+            <DialogContent>
+                <Box sx={{ padding: 1 }}>
+                    <DialogContentText>
+                        {intl.formatMessage({
+                            id: 'restoreNodesText',
+                        })}
+                    </DialogContentText>
+                </Box>
+                {isLoading && (
+                    <LoaderWithOverlay
+                        color="inherit"
+                        loaderSize={70}
+                        isFixed={false}
+                        loadingMessageText={'LoadingRemoteData'}
+                    />
+                )}
+                {!isLoading && nodes && (
+                    <FormControl
+                        sx={{ paddingLeft: '20px' }}
+                        component="fieldset"
+                    >
+                        <RadioGroup name="nodes-to-restore-selection">
```

Comment:
On the feature mockup it looks like checkbox were initially intended, was this change intented ?

Reply:
yes it was discussed and a single choice is what is wanted -> radio